package.nix

```diff
@@ -1,196 +1,207 @@
+{
+  nixos-generators,
+  nixpkgs,
+  # pkgs
+  lib,
+  mount,
+  umount,
+  # configuration
+  linuxSystem,
+  debug ? false,
+  extraConfig ? {},
+  onDemand ? false, # enable launchd socket activation
+  withRosetta ? true,
 }:
-nixos-generators.nixosGenerate (
-let
-  inherit (import ./constants.nix)
-    linuxHostName
-    linuxUser
-
-    sshHostPrivateKeyFileName
-    sshUserPublicKeyFileName
-
-    debug
-    ;
-  imageFormat = "qcow-efi"; # must match `vmYaml.images.location`s extension
-
-  sshdKeys = "sshd-keys";
-  sshDirPath = "/etc/ssh";
-  sshHostPrivateKeyFilePath = "${sshDirPath}/${sshHostPrivateKeyFileName}";
-
-in {
-  format = imageFormat;
-
-  modules = [ {
-    boot = {
-      kernelParams = [ "console=tty0" ];
-
-      loader = {
-        efi.canTouchEfiVariables = true;
-        systemd-boot.enable = true;
-      };
-    };
-
-    documentation.enable = false;
-
-    fileSystems = {
-      "/".options = [ "discard" "noatime" ];
-      "/boot".options = [ "discard" "noatime" "umask=0077" ];
-    };
-
-    networking.hostName = linuxHostName;
-
-    nix = {
-      channel.enable = false;
-      registry.nixpkgs.flake = nixpkgs;
-
-      settings = {
-        auto-optimise-store = true;
-        experimental-features = [ "flakes" "nix-command" ];
-        min-free = "5G";
-        max-free = "7G";
-        trusted-users = [ linuxUser ];
-      };
-    };
-
-    security = {
-      sudo = {
-        enable = debug;
-        wheelNeedsPassword = !debug;
-      };
-    };
-
-    services = {
-      getty = lib.optionalAttrs debug { autologinUser = linuxUser; };
-
-      logind = lib.optionalAttrs onDemand {
-        extraConfig = ''
-          IdleAction=poweroff
-          IdleActionSec=3h
-        '';
-      };
-
-      openssh = {
-        enable = true;
-        hostKeys = []; # disable automatic host key generation
-
-        settings = {
-          HostKey = sshHostPrivateKeyFilePath;
-          PasswordAuthentication = false;
-        };
-      };
-    };
-
-    system = {
-      disableInstallerTools = true;
-      stateVersion = "24.05";
-    };
-
-    # macOS' Virtualization framework's virtiofs implementation will grant any guest user access
-    # to mounted files; they always appear to be owned by the effective UID and so access cannot
-    # be restricted.
-    # To protect the guest's SSH host key, the VM is configured to prevent any logins (via
-    # console, SSH, etc) by default.  This service then runs before sshd, mounts virtiofs,
-    # copies the keys to local files (with appropriate ownership and permissions), and unmounts
-    # the filesystem before allowing SSH to start.
-    # Once SSH has been allowed to start (and given the guest user a chance to log in), the
-    # virtiofs must never be mounted again (as the user could have left some process active to
-    # read its secrets).  This is prevented by `unitconfig.ConditionPathExists` below.
-    systemd.services."${sshdKeys}" =
+with lib;
```

Comment:
Likewise module.nix's `with lib` comment.

Reply:
Resolved in https://github.com/cpick/nix-rosetta-builder/commit/6d744549534fad259688cf32aec6a55bf7037ad3